src/components/table/table.tsx

```diff
@@ -143,6 +144,28 @@ class Table<IDataCoordinates = any> extends React.Component<ITableProps<IDataCoo
     );
   }
 
+  static getDerivedStateFromProps(props: ITableProps, state: IState) {
+    const {
+      rows,
+      onOpenedTreesUpdate,
+      virtualizerProps: { fixedRows },
+    } = props;
+
+    const newOpenedTrees = syncOpenedTreesWithRows(state.openedTrees, rows);
+    if (onOpenedTreesUpdate) {
+      onOpenedTreesUpdate(newOpenedTrees);
+    }
+
+    const newIndexesMapping = getAllIndexesMap(newOpenedTrees, rows);
+    return {
+      openedTrees: newOpenedTrees,
+      indexesMapping: newIndexesMapping,
+      rowsLength: getRowslength(newOpenedTrees, rows),
+      columnsIndexesIdsMapping: rows[0] ? getIndexesIdsMapping(rows[0].cells) : {},
+      fixedRowsIndexes: getFixedRowsIndexes(newOpenedTrees, newIndexesMapping.relative, rows, fixedRows),
+    };
+  }
```

Comment:
I move the logic to sync state from `componentDidUpdate` to here because our state could be out of sync for a render. This is what is happening:

Before (syncing state in componentDidUpdate):
 1. rows prop change
 2. render (state out of sync)
 3. componentDidUpdate => we update the state
 4. rerender with new state
 5. everything in sync
 
 Problem: at step 2 we are out of sync, so I had to put a guard check (hence this change https://github.com/PricemoovOSS/react-table/pull/6#discussion_r1338916474)
 
 Moving this logic in `getDerivedStateFromProps`:
 
  1. rows prop change
  2. getDerivedStateFromProps is called to sync the state
  3. render (everything in sync)
  
  So this prevents us from having double rerenders and state is always in sync at all times